File: src/test/java/io/binac/leetcode/FlattenBinaryTreeToLinkedListTest.java
package io.binac.leetcode;

import io.binac.leetcode.util.TreeNode;
import io.binac.leetcode.util.Trees;
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.*;

class FlattenBinaryTreeToLinkedListTest {
    private final FlattenBinaryTreeToLinkedList.Solution1 solution1 = new FlattenBinaryTreeToLinkedList.Solution1();

    @Test
    void test1() {
        TreeNode root = Trees.asTree(1, 2, 5, 3, 4, null, 6);
        solution1.flatten(root);
        TreeNode expected = Trees.asTree(1, null, 2, null, 3, null, 4, null, 5, null, 6);
        assertTrue(Trees.equals(expected, root));
    }
}

File: cpp/unittests/FlattenBinaryTreeToLinkedListTest.cpp
#include "FlattenBinaryTreeToLinkedList.h"
#include "gtest/gtest.h"

using namespace lcpp;

TEST(FlattenBinaryTreeToLinkedList, Solution114_1) {
  const auto Root = newTree({1, 2, 5, 3, 4, NTNode, 6}), Expected =
      newTree({1, NTNode, 2, NTNode, 3, NTNode, 4, NTNode, 5, NTNode, 6});
  Solution114_1().flatten(Root);
  EXPECT_EQ(*Expected, *Root);
  deleteTree(Root);
  deleteTree(Expected);
}